File: Python/Euler009.py
#!/usr/bin/env python3
from __future__ import division, print_function
from math import sqrt, floor, ceil
import sys
if sys.version.startswith('2'):
    range = xrange

def main():
    target = 1000
    for a in range(3, int(ceil((target - 1) / 3))):
        for b in range(a + 1, int(ceil(target - a / 2))):
            c = sqrt(a*a + b*b)
            if c == floor(c):
                c = int(c)
                s = a + b + c

                factor, mod = divmod(target, s)
                if mod is 0:
                    aFinal = factor * a
                    bFinal = factor * b
                    cFinal = factor * c

                    print(aFinal * bFinal * cFinal)
                    return
                else:
                    break

if __name__ == '__main__':
    main()

File: Ruby/Euler009.rb
#!/usr/bin/env ruby

def main

  target = 1000
  for a in (3..((target - 1) / 3).ceil)
    for b in ((a + 1)..(target - a / 2).ceil)
      c = Math.sqrt(a*a + b*b)
      if c == c.floor
        c = c.to_i
        s = a + b + c

        factor, mod = target.divmod(s)
        if mod == 0
          aFinal = factor * a
          bFinal = factor * b
          cFinal = factor * c

          puts (aFinal * bFinal * cFinal)
          return
        else
          break
        end
      end
    end
  end
end

if __FILE__ == $0
  main
end
